src/main/provider/index.tsx

```diff
@@ -50,16 +50,22 @@ export const NumeratorProvider: React.FC<NumeratorProviderProps> = ({
   const fetchPollingFeatureFlag = useCallback(async () => {
     try {
       const result = await numeratorClient.fetchPollingFlag(defaultContextValues, currentEtag);
-      setCurrentEtag(result.etag);
-      const newCache = result.flags.reduce(
-        (acc, flag) => {
-          acc[flag.key] = flag;
-          return acc;
-        },
-        {} as Record<string, any>,
-      );
-      setCacheFlags(newCache);
-      updateListeners.forEach((listener) => listener(newCache)); // Notify all update listeners
+      if (result.flags) {
+        // 200 OK
+        setCurrentEtag(result.etag);
+        const newCache = result.flags.reduce(
+          (acc, flag) => {
+            acc[flag.key] = flag;
+            return acc;
+          },
+          {} as Record<string, any>,
+        );
+        setCacheFlags(newCache);
+        updateListeners.forEach((listener) => listener(newCache)); // Notify all update listeners
+      } else {
+        // 304 NOT MODIFIED
+        updateListeners.forEach((listener) => listener(cacheFlags)); // Notify all update listeners
```

Comment:
I think, we should only push events when we have updated new data, in case 304 we can skip this event. We don't want to push event spam without update the client